pandas/tests/test_categorical.py

```diff
@@ -3651,6 +3651,12 @@ def test_concat_categorical(self):
 
         tm.assert_frame_equal(df_expected, df_concat)
 
+    def test_density(self):
+        s1 = Series(list('aabbcccc')).astype('category')
+        self.assertEqual(0.375, s1.cat.density)
+
+        s2 = Series([]).astype('category')
+        tm.assert_numpy_array_equal(np.nan, s2.cat.density)
```

Comment:
Quick question on this change: np.nan does not equal np.nan so this test will fail. Am I correct in thinking that s2.cat.density should equal np.nan?